api/team.go

```diff
@@ -708,6 +709,19 @@ func findTeamByName(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func getTeamByName(c *Context, w http.ResponseWriter, r *http.Request) {
+	params := mux.Vars(r)
+	teamname := params["team_name"]
+
+	if result := <-Srv.Store.Team().GetByName(teamname); result.Err != nil {
+		c.Err = result.Err
+		return
+	} else {
```

Comment:
Also once you have the team I think you need to do a permissions check like this:

```
if team.Type != model.TEAM_OPEN && c.Session.GetTeamByTeamId(team.Id) == nil {
	if !HasPermissionToContext(c, model.PERMISSION_MANAGE_SYSTEM) {
		return
	}
}
```